Added several more domains to cubecraft.xml.

marcoslater opened the pull request:
Added several HTTPS domains for cubecraft.net.

Added status page redirect for HTTPS on there as well.

marcoslater commented:
Seems CI is having issues connect to the git repos it needs to complete the checks, hmm.

marcoslater commented on src/chrome/content/rules/cubecraft.xml:
```diff
@@ -1,8 +1,14 @@
+        <target host="stats.cubecraft.net" />
+        <target host="status.cubecraft.net" />
+
+        <rule from="^http://status\.cubecrtaft\.net/"
```
Ouch! Thanks for noticing that! :)

jeremyn commented:
@marcoslater Are you still interested in working on this?

marcoslater commented:
@jeremyn Thanks for reminding of this. I manage that domain, so I do intend to keep updating it. Let me add the latest subdomain to the mix. 

There's only one subdomain keeping us from site-wide HSTS, sadly.

jeremyn commented:
No problem, thanks for working with us. 

This ruleset has multiple problems. Please update it to look like this:

```xml
<!--
	Invalid certificate:
		status.cubecraft.net

-->
<ruleset name="CubeCraft">
	<target host="cubecraft.net" />
	<target host="www.cubecraft.net" />
	<target host="appeals.cubecraft.net" />
	<target host="bb.cubecraft.net" />
	<target host="cdn.cubecraft.net" />
	<target host="jobs.cubecraft.net" />
	<target host="reports.cubecraft.net" />
	<target host="staff.cubecraft.net" />
	<target host="status.cubecraft.net" />
	<target host="store.cubecraft.net" />
	<target host="store-assets.cubecraft.net" />

	<securecookie host=".+" name=".+" />

	<rule from="^http://status\.cubecraft\.net/"
		to="https://cubecraft.statuspage.io/" />

	<rule from="^http:" to="https:" />
</ruleset>
```

This fixes the following problems:
* Indents with tabs instead of spaces to match our style guide, see [here](https://github.com/EFForg/https-everywhere/blob/master/ruleset-style.md).
* Adds a comment describing the problem with https://status.cubecraft.net . 
* Removes the wildcard `target`, which we discourage. See our style guide linked above.
* Removes `test`s, which aren't needed here without the wildcard target. The `test`s also had multiple problems: `test`s should almost always be `http`, not `https`; `test`s should match `target`s, so the https://cubecraft.statuspage.io test [here](https://github.com/marcoslater/https-everywhere/blob/2a1114bda548fd11c489c0327f82a92c024891d6/src/chrome/content/rules/cubecraft.xml#L8) wasn't appropriate.
* Adds `target`s.
* Sorts `target`s in our preferred way: in alphabetical order starting from the top level domain at the right reading left, moving `^` and `www` to the top of their group. For example:
```
example.com
www.example.com
a.example.com
www.a.example.com
b.a.example.com
b.example.com
example.net
www.example.net
a.example.net
```
* Adds general `securecookie`.
* Fixes misspelling in the rewrite rule [here](https://github.com/marcoslater/https-everywhere/blob/2a1114bda548fd11c489c0327f82a92c024891d6/src/chrome/content/rules/cubecraft.xml#L10).

The remaining concerns are:

- [ ] Is https://swc.cubecraft.net something you care about? It's getting a Cloudflare error with both HTTPS and plain HTTP, so I'm not sure if it should be added or not. We don't normally add targets that are broken when we check them.
- [ ] We like to [squash-and-merge](https://help.github.com/articles/about-pull-request-merges/#squash-and-merge-your-pull-request-commits) rulesets, which means we combine and rewrite commits in your name. Is that all right with you for this and any other pull requests you submit to us?
- [ ] Also, when I squash-and-merge, I'd like to delete your individual commit messages. Is that all right?

marcoslater commented:
Didn't realise I screwed that one up that badly. 

Changed it to reflect your version, also added api.cubecraft.net, as that is used in various places on those pages. (once logged in, at least.) 

swc.cubecraft.net was delegated to a third party, clearly they do not intend to keep hold of that. Removed it from our DNS. (Query NS directly if you wish to verify this.) 

Feel free to modify my commits however you like.

jeremyn commented:
Thanks, that's fine.

For https://api.cubecraft.net , is there a URL for this subdomain we can access without logging in that gets a `200` or `3xx`? I don't like to include `target`s without at least one working domain (see the discussion in  https://github.com/EFForg/https-everywhere/issues/7662).

jeremyn commented:
Also don't worry about "screwing it up". Some of my first pull requests here had trouble too. Our documentation could definitely be improved.

marcoslater commented:
@jeremyn RE: API not having a 200, check / now. :)

jeremyn commented:
Thanks, merged.

If/when you get everything HSTS preloaded, feel free to submit another pull request to delete this ruleset. Generally we want all the domains in a ruleset to be preloaded in the stable versions of Firefox, Chrome/ium, and Tor before we delete that ruleset. See https://github.com/EFForg/https-everywhere/issues/7126 for an extended discussion.

marcoslater commented:
I am working towards that goal with our last provider, which has a subdomain (status) that does not have HTTPS capabilities. Once done, I can enable HSTS preloading and let you know once done.

marcoslater commented:
@jeremyn cubecraft.net is now pending submission to preload list. :)

jeremyn commented:
Great!
